src/WalkingModule/src/Module.cpp

```diff
@@ -521,27 +638,63 @@ bool WalkingModule::updateModule()
         // check desired planner input
         yarp::sig::Vector* desiredUnicyclePosition = nullptr;
         desiredUnicyclePosition = m_desiredUnyciclePositionPort.read(false);
-        if(desiredUnicyclePosition != nullptr)
-            if(!setPlannerInput((*desiredUnicyclePosition)(0), (*desiredUnicyclePosition)(1)))
+        if(m_useStepAdaptation)
+        {
+            if(desiredUnicyclePosition != nullptr)
             {
-                yError() << "[WalkingModule::updateModule] Unable to set the planner input";
-                return false;
+                if(!setPlannerInput((*desiredUnicyclePosition)(0), 0.0))
```

Comment:
Is there any plan to remove this limitation?

Reply:
Yes I remove this.